### ISSUE TITLE
android-security-crypto is now deprecated

### ISSUE THREAD

**Issue by barnhill** [LabMobi/LabEncryptedStorage#11]

**Comment by harri35** [comment]
Yes, indeed, as per note at https://developer.android.com/privacy-and-security/cryptography#deprecated-functionality
As soon as Google has an official recommendation about the successor we can plan the update to this library.
For now, it seems android-security-crypto continues to work

**Comment by kuldar-daniel** [comment]
Android policy security high com.google.crypto.tink : tink-android : 1.8.0

A security vulnerability has been discovered in the dependencies for said crypto library: [cve-2024-7254](https://nvd.nist.gov/vuln/detail/cve-2024-7254)

Dependency resolution:
```
<dependency id="mobi.lab.labencryptedstorage:labencryptedstorage:0.10.3">
   <dependency id="androidx.security:security-crypto:1.1.0-alpha06">
      <dependency id="androidx.collection:collection:1.4.2"/>
         <dependency id="androidx.annotation:annotation:1.9.1"/>
         <dependency id="com.google.crypto.tink:tink-android:1.8.0">
            <dependency id="com.google.code.gson:gson:2.12.0">
              <dependency id="com.google.errorprone:error_prone_annotations:2.36.0"/>
            </dependency>
         </dependency>
</dependency>
```

**Comment by kuldar-daniel** [comment]
https://developer.android.com/jetpack/androidx/releases/security it also looks like non-alpha versions of the library have been released this year, perhaps the underlying dependencies are updated in 1.1.0?

**Comment by harri35** [comment]
@kuldar-daniel 

> https://developer.android.com/jetpack/androidx/releases/security it also looks like non-alpha versions of the library have been released this year, perhaps the underlying dependencies are updated in 1.1.0?

Indeed, interesting. Maybe they made a bugfix there or smth:
> This library was last updated on: July 30, 2025

Are you in a hurry to figure this out? If the updated dependency solves this issue, then make a PR, I can review and merge it asap then.  I recently updated this project to the new OSSRH publishing plugin, I can easily publish a newer version of the library
